core/templates/pages/admin-page/features-tab/admin-features-tab.component.ts

```diff
@@ -0,0 +1,249 @@
+      })
+    );
+  }
+
+  addNewRuleToBottom(feature: PlatformParameter): void {
+    feature.rules.push(
+      this.ruleFactory.createFromBackendDict({
+        filters: [{
+          type: PlatformParameterFilterType.ServerMode,
+          conditions: [['=', ServerMode.Dev.toString()]]
+        }],
+        value_when_matched: false
+      })
+    );
+  }
+
+  addNewFilter(rule: PlatformParameterRule): void {
+    rule.filters.push(
+      this.filterFactory.createFromBackendDict({
+        type: PlatformParameterFilterType.ServerMode,
+        conditions: []
+      })
+    );
+  }
+
+  addNewCondition(filter: PlatformParameterFilter): void {
+    const context = this.filterTypeToContext[filter.type];
+    filter.conditions.push([
+      context.operators[0],
+      context.options ? context.options[0] : ''
+    ]);
+  }
+
+  removeRule(feature: PlatformParameter, ruleIndex: number): void {
+    feature.rules.splice(ruleIndex, 1);
+  }
+
+  removeFilter(rule: PlatformParameterRule, filterIndex: number): void {
+    rule.filters.splice(filterIndex, 1);
+  }
+
+  removeCondition(
+      filter: PlatformParameterFilter, conditionIndex: number): void {
+    filter.conditions.splice(conditionIndex, 1);
+  }
+
+  moveRuleUp(feature: PlatformParameter, ruleIndex: number): void {
+    const rule = feature.rules[ruleIndex];
+    this.removeRule(feature, ruleIndex);
+    feature.rules.splice(ruleIndex - 1, 0, rule);
+  }
+
+  moveRuleDown(feature: PlatformParameter, ruleIndex: number): void {
+    const rule = feature.rules[ruleIndex];
+    this.removeRule(feature, ruleIndex);
+    feature.rules.splice(ruleIndex + 1, 0, rule);
+  }
+
+  async updateFeatureRulesAsync(feature: PlatformParameter): Promise<void> {
+    if (this.adminTaskManager.isTaskRunning()) {
+      return;
+    }
+    const commitMessage = this.windowRef.nativeWindow.prompt(
+      'This action is irreversible. If you insist to proceed, please enter ' +
+      'the commit message for the update',
+      `Update feature '${feature.name}'.`
+    );
+    if (commitMessage === null) {
+      return;
+    }
+
+    try {
+      this.adminTaskManager.startTask();
+
+      await this.apiService.updateFeatureFlag(
+        feature.name, commitMessage, feature.rules);
+
+      this.setStatusMessage('Saved successfully.');
+    } catch (e) {
+      if (e.error && e.error.error) {
+        this.setStatusMessage(`Update failed: ${e.error.error}`);
+      } else {
+        this.setStatusMessage('Update failed.');
+      }
+    } finally {
+      this.adminTaskManager.finishTask();
+    }
+  }
+
+  clearChanges(featureFlag: PlatformParameter): void {
+    if (!this.windowRef.nativeWindow.confirm(
+      'This will revert all changes you made. Are you sure?')) {
+      return;
+    }
+    const backup = this.featureFlagNameToBackupMap.get(featureFlag.name);
+    featureFlag.rules = backup.rules;
+  }
+
+  onFilterTypeSelectionChanged(filter: PlatformParameterFilter): void {
```

Comment:
Perhaps this should be ``removeTopFilterCondition`` or similar?

Reply:
This is called when the type of a filter is changed, in such cases all existsing conditions (operators, values) are not applicable any longer, so this method clear all conditions of that filter.
Renamed it to `clearFilterConditions` to make it clearer.